handlers/GuildSettings.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+const path = require('path');
+
+const guildConfigsPath = path.join(__dirname, '../config/guilds');
+
+exports.init = () => {
+  const guildOverrides = {};
+
+  fs.readdir(guildConfigsPath, (err, files) => {
+    if (err) throw err;
+
+    files.forEach((file) => {
+      if (file.endsWith('.json')) {
+        const guildId = file.slice(0, file.length - 5);
+        const guildConfig = JSON.parse(fs.readFileSync(path.resolve(guildConfigsPath, `./${file}`), 'utf-8'));
+
+        console.log(guildConfig);
+
+        guildOverrides[guildId] = guildConfig;
+      }
+    });
+  });
+
+  return guildOverrides;
+};
+
+exports.getGuildCommandPrefix = (bot, message) => {
+  let id = null;
+
+  if (message && message.guild) {
+    ({ id } = message.guild);
+  }
+  if (id) {
+    if (bot.guildOverrides[id] && bot.guildOverrides[id].prefix) {
+      return bot.guildOverrides[id].prefix;
+    }
+  }
+
+  return bot.prefix;
+};
+
+// exports.update = (bot, id, key, value) => {
```

Comment:
ways in which I envision this...

- Checks for `GUILD_ID.json` in `config/guilds/`
  - if `EXISTS` then
    - Load data from the JSON
    - Edit the value in `<SCOPE>.<KEY>` (safe check for the `<SCOPE>` first)
    - Save data to the JSON file
    - Update this in `bot.guildOverrides` as well
  - else (`EXISTS === false`)
    - Create new `GUILD_ID.json` file
    - Edit in the appropriate `<SCOPE>.<KEY>.<VALUE>` incoming from command
    - Save data to the JSON file
    - Update this in `bot.guildOverrides` as well
- Reply back the following...
  - New guild config was created, and the value `<SCOPE>.<KEY>.<VALUE>` was written
  - Existing guild config was edited, and the value `<SCOPE>.<KEY>.<VALUE>` was written
- Help menus should show available settings that can be set, per-server
  - Dynamically allowing all `settings.json` values to be per-server will be a headache, let's just allow specific stuff at first
  - Think of a good way to showcase the available `<SCOPE>`s, `<KEY>`s, and their appropriate types for setting the `<VALUE>`
    - This might require type-checking, parsing and error handling/reporting back to user for bad command usage...